src/WinGetServer/Utils.cpp

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "Utils.h"
+#pragma warning( push )
+#pragma warning ( disable : 6001 6388 6553)
+#include <wil/resource.h>
+#pragma warning( pop )
+#include <processthreadsapi.h>
+#include <sddl.h>
+#include <vector>
+
+unsigned char* GetUCharString(const std::string& str)
+{
+    return reinterpret_cast<unsigned char*>(const_cast<char*>(str.c_str()));
+}
+
+std::string GetUserSID()
+{
+    HANDLE hToken = NULL;
+    THROW_LAST_ERROR_IF(!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken));
+
+    DWORD dwBufferSize = 0;
+    THROW_LAST_ERROR_IF(!GetTokenInformation(hToken, TokenUser, NULL, 0, &dwBufferSize) && GetLastError() != ERROR_INSUFFICIENT_BUFFER);
+
+    std::vector<BYTE> buffer;
+    buffer.resize(dwBufferSize);
+    PTOKEN_USER pTokenUser = reinterpret_cast<PTOKEN_USER>(&buffer[0]);
+
+    THROW_LAST_ERROR_IF(!GetTokenInformation(hToken, TokenUser, pTokenUser, dwBufferSize, &dwBufferSize));
+    THROW_LAST_ERROR_IF(!IsValidSid(pTokenUser->User.Sid));

```

Comment:
>THROW_LAST_ERROR_IF(!IsValidSid(pTokenUser->User.Sid)); [](http://example.com/codeflow?start=4&length=55)

the api doc said do not call GetLastError()
[https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-isvalidsid](https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-isvalidsid)